### DISCUSSION TITLE
[BUG]: unable to get private repos while using octokit

### DISCUSSION THREAD

**Discussion by statusunknown418** [octokit/octokit.js#2420] [Q&A]
### What happened?

doing this works with curl but not octokit which seems weird, this won't show private repos even while having authorization

```ts
const { data } = await octokit.request("GET /search/repositories", {
        q: `org:${selectedOrganization}`,
        headers: {
          authorization: `Bearer ${token}`,
        },
      });
```

Update:

Looking at the request I can see that gh is not authenticating the user, no idea why, getting unauthenticated rate limits
<img width="1054" alt="Screenshot 2023-03-27 at 19 35 36" src="https://user-images.githubusercontent.com/74371045/228096857-3e126719-76a2-4c84-921f-982c72ff5fa8.png">

### Versions

Octokit 2.0.14

### Relevant log output

```shell
already in photos
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JayKim88** [comment]
Hi, @AlvaroAquijeDiaz 
I hope this code below helps you-.

```ts
// Octokit.js
// https://github.com/octokit/core.js#readme
const octokit = new Octokit({
  auth: 'YOUR-TOKEN'
})

await octokit.request('GET /repos/{owner}/{repo}', {
  owner: 'OWNER',
  repo: 'REPO',
  headers: {
    'X-GitHub-Api-Version': '2022-11-28'
  }
})
```
Ref: 
- https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository
- https://github.com/octokit/octokit.js#rest-api

And another code below is an example I'm using to get specific workflow data.
This is just for a reference different from your purpose. Try out :)

```ts
      const octokit = new res.Octokit({
        auth: gh_token,
      });

      const gitActionData = await octokit.request(
        'GET /repos/{owner}/{repo}/actions/workflows/{workflow_id}/runs',
        {
          owner: 'owner-name',
          repo: 'repo-name',
          workflow_id: 'workflow_id',
          per_page: 1,
          page: 1,
          status: 'completed',
        }
      );
```

**Reply by huperniketes** [reply]
I'm unable to get private repos thru curl, but the API does return public repos.

What permissions do I need to have granted to have private repos returned by “GET /user/repos?type=private”?

**Reply by Tazi0** [reply]
@huperniketes did you find a solution to your problem yet?

**Comment by HectorGitt** [comment]
Same problem

**Comment by HectorGitt** [comment]
Was able to find a solution, seems like you need to install the app in the users settings manually.
